src/commit.c

```diff
@@ -382,41 +383,53 @@ int git_commit_amend(
 	return error;
 }
 
-int git_commit__parse(void *_commit, git_odb_object *odb_obj)
+int git_commit__parse_commit(git_commit *commit, git_odb_object *odb_obj, git_commit__parse_flags flags)
 {
-	git_commit *commit = _commit;
 	const char *buffer_start = git_odb_object_data(odb_obj), *buffer;
 	const char *buffer_end = buffer_start + git_odb_object_size(odb_obj);
-	git_oid parent_id;
 	size_t header_len;
 	git_signature dummy_sig;
 
-	buffer = buffer_start;
+	assert(commit && odb_obj);
 
-	/* Allocate for one, which will allow not to realloc 90% of the time  */
-	git_array_init_to_size(commit->parent_ids, 1);
-	GITERR_CHECK_ARRAY(commit->parent_ids);
+	buffer = buffer_start;
 
 	/* The tree is always the first field */
-	if (git_oid__parse(&commit->tree_id, &buffer, buffer_end, "tree ") < 0)
-		goto bad_buffer;
+	if (flags & GIT_COMMIT_PARSE_ALL) {
+		if (git_oid__parse(&commit->tree_id, &buffer, buffer_end, "tree ") < 0)
+			goto bad_buffer;
+	} else {
+		buffer += strlen("tree ") + GIT_OID_HEXSZ + 1;
+	}
+
+	if (flags & (GIT_COMMIT_PARSE_ALL|GIT_COMMIT_PARSE_QUICK)) {
```

Comment:
Can't this condition just be removed? You're doing it for all flags anyway, so no real need to check for those flags.

Reply:
I did that mostly for completeness — that makes it easier to spot the differences between 
`ALL` and `QUICK`, and also (somewhat) future-proof. I can remove though, if that's preferred.